terraform/modules/relay-app/main.tf

```diff
@@ -0,0 +1,365 @@
+    google_project_service.cloudprofiler,
+    google_project_service.cloudtrace,
+    google_project_service.servicenetworking,
+    google_project_iam_member.artifacts,
+    google_project_iam_member.logs,
+    google_project_iam_member.errors,
+    google_project_iam_member.metrics,
+    google_project_iam_member.service_management,
+    google_project_iam_member.cloudtrace,
+  ]
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+# Create health checks for the application ports
+resource "google_compute_health_check" "port" {
+  project = var.project_id
+
+  name = "${local.application_name}-${var.health_check.name}"
+
+  check_interval_sec  = var.health_check.check_interval_sec != null ? var.health_check.check_interval_sec : 5
+  timeout_sec         = var.health_check.timeout_sec != null ? var.health_check.timeout_sec : 5
+  healthy_threshold   = var.health_check.healthy_threshold != null ? var.health_check.healthy_threshold : 2
+  unhealthy_threshold = var.health_check.unhealthy_threshold != null ? var.health_check.unhealthy_threshold : 2
+
+  log_config {
+    enable = false
+  }
+
+  http_health_check {
+    port = var.health_check.port
+
+    host         = var.health_check.http_health_check.host
+    request_path = var.health_check.http_health_check.request_path
+    response     = var.health_check.http_health_check.response
+  }
+}
+
+# Use template to deploy zonal instance group
+resource "google_compute_instance_group_manager" "application" {
+  for_each = var.instances
+
+  project = var.project_id
+
+  name = "${local.application_name}-group"
+
+  base_instance_name = local.application_name
+  zone               = each.key
+
+  target_size = each.value.replicas
+
+  wait_for_instances        = true
+  wait_for_instances_status = "STABLE"
+
+  version {
+    instance_template = google_compute_instance_template.application[each.key].self_link
+  }
+
+  named_port {
+    name = "stun"
+    port = 3478
+  }
+
+  auto_healing_policies {
+    initial_delay_sec = var.health_check.initial_delay_sec
+
+    health_check = google_compute_health_check.port.self_link
+  }
+
+  update_policy {
+    type           = "PROACTIVE"
+    minimal_action = "REPLACE"
+
+    max_unavailable_fixed = 1
+    max_surge_fixed       = max(1, each.value.replicas - 1)
+  }
+
+  depends_on = [
+    google_compute_instance_template.application
+  ]
+}
+
+# Define a security policy which allows to filter traffic by IP address,
+# an edge security policy can also detect and block common types of web attacks
+resource "google_compute_security_policy" "default" {
+  project = var.project_id
+
+  name = local.application_name
+
+  rule {
+    action   = "allow"
+    priority = "2147483647"
+
+    match {
+      versioned_expr = "SRC_IPS_V1"
+
+      config {
+        src_ip_ranges = ["*"]
+      }
+    }
+
+    description = "default allow rule"
+  }
+}
+
+# Open ports for the web
+resource "google_compute_firewall" "http" {
```

Comment:
```suggestion
# Open ports for STUN/TURN
resource "google_compute_firewall" "stun-turn" {
```

Maybe?

Reply:
Yeah, this module started as a copy of Elixir one, hence all the naming artifacts